ksql-engine/src/test/java/io/confluent/ksql/function/BlacklistTest.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.ksql.function;
+
+import com.google.common.collect.ImmutableList;
+
+import org.apache.kafka.test.TestUtils;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.List;
+
+import static junit.framework.TestCase.assertFalse;
+import static junit.framework.TestCase.assertTrue;
+
+public class BlacklistTest {
+
+  private File blacklistFile;
+
+  @Before
+  public void before() throws IOException {
+     blacklistFile = TestUtils.tempFile();
+  }
+
+  @Test
+  public void shouldBlackListAllInPackage() throws IOException {
+    writeBlacklist(ImmutableList.of("java.lang"));
+    final Blacklist blacklist = new Blacklist(this.blacklistFile);
+    assertTrue(blacklist.test("java.lang.Class"));
+    assertFalse(blacklist.test("java.util.List"));
+  }
+
+  @Test
+  public void shouldBlackListClassesMatching() throws IOException {
+    writeBlacklist(ImmutableList.of("java.lang.Process"));
+    final Blacklist blacklist = new Blacklist(this.blacklistFile);
+    assertTrue(blacklist.test("java.lang.Process"));
+    assertTrue(blacklist.test("java.lang.ProcessBuilder"));
+    assertTrue(blacklist.test("java.lang.ProcessEnvironment"));
+    assertFalse(blacklist.test("java.lang.Class"));
+  }
+
+  @SuppressWarnings("ResultOfMethodCallIgnored")
+  @Test
+  public void shouldNotBlacklistAnythingIfFailsToLoadFile() {
```

Comment:
We do expect to provide a default blacklist file which is required for safe functioning, right? If so, would it rather make sense to blacklist all udfs if the file is missing?

Reply:
Yeah, we could make it so that if the file is missing we blacklist. If the file is there, but empty, it means allow all.